core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -2043,6 +2045,52 @@ List<SqlNode> getListOfCaseOperands(SqlCase sqlCase) {
       return operandList;
     }
 
+    RexNode getRexInputRefFromCondition(RexNode condition) {
+      if (condition instanceof RexInputRef) {
+        return condition;
+      } else if (condition instanceof RexCall) {
+        for (RexNode operand : ((RexCall) condition).getOperands()) {
+          if (operand instanceof  RexLiteral) {
+            continue;
+          } else {
+            return getRexInputRefFromCondition(operand);
+          }
+        }
+      }
+      return condition;
+    }
+
+    /** Returns whether an Analytical Function is present in filter condition. */
+    private boolean hasAnalyticalFunctionPresentInFilter(RelNode rel) {
```

Comment:
Can we rename it to `hasAnalyticalFunctionInFilter`?

Reply:
Done, renamed.